Core/CategoriesManager.cs

```diff
@@ -1,36 +1,34 @@
-﻿using Newtonsoft.Json;
-using System;
+﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
 using System.Text.RegularExpressions;
 
 namespace Core
 {
     public static class CategoriesManager
     {
-		public static Dictionary<string, Func<Transaction, bool>> Load()
+		public static Dictionary<string, Func<Transaction, bool>> BuildFilters(List<Category> categories)
 		{
-			//TODO: load from user provided file
-			if (true)
+			var categoryFilters = new Dictionary<string, Func<Transaction, bool>>();
+			
+			foreach (var c in categories)
 			{
-				return new Dictionary<string, Func<Transaction, bool>>();
+				categoryFilters.Add(c.Name, t => c.Rules.Any(r => Regex.IsMatch(t.CardNumber, r.CardNumber)
+											&& Regex.IsMatch(t.Description, r.Description)
+											&& Regex.IsMatch(t.Amount.ToString().ToLower(), r.Amount)
+											&& r.Amount[1..] is var ch
+											&& r.Amount[0] switch
+											{
+												'>' => t.Amount.Amount > int.Parse(ch),
+												'<' => t.Amount.Amount < int.Parse(ch),
+												'=' => t.Amount.Amount == int.Parse(ch),
+												'*' => true,
+												_ => throw new NotImplementedException()
```

Comment:
Codacy found an issue: [Implement this method or throw 'NotSupportedException' instead.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=5836065)